src/app/cluster/cluster-details/constraints/component.ts

```diff
@@ -0,0 +1,183 @@
+  @Input() cluster: Cluster;
+  @Input() projectID: string;
+  @Input() isClusterRunning: boolean;
+  constraints: Constraint[] = [];
+  settings: UserSettings;
+  dataSource = new MatTableDataSource<Constraint>();
+  displayedColumns: string[] = ['constraintName', 'constraintTemplate', 'match', 'violations', 'actions'];
+  toggledColumns: string[] = ['violationDetails'];
+  isShowDetails = [];
+  @ViewChild(MatSort, {static: true}) sort: MatSort;
+  paginator: MatPaginator;
+  @ViewChild(MatPaginator)
+  set matPaginator(mp: MatPaginator) {
+    this.paginator = mp;
+    if (this.paginator && this.settings) {
+      this.paginator.pageSize = this.settings.itemsPerPage;
+    }
+    this.dataSource.paginator = this.paginator;
+    this._cdr.detectChanges();
+  }
+
+  private _unsubscribe = new Subject<void>();
+
+  constructor(
+    private readonly _opaService: OPAService,
+    private readonly _userService: UserService,
+    private readonly _cdr: ChangeDetectorRef,
+    private readonly _matDialog: MatDialog,
+    private readonly _notificationService: NotificationService
+  ) {}
+
+  ngOnInit(): void {
+    this.dataSource.data = this.constraints;
+    this.dataSource.sort = this.sort;
+    this.dataSource.paginator = this.paginator;
+    this.sort.active = 'constraintName';
+    this.sort.direction = 'asc';
+
+    this._opaService
+      .constraints(this.projectID, this.cluster.id)
+      .pipe(takeUntil(this._unsubscribe))
+      .subscribe(constraints => {
+        this.constraints = constraints;
+        this.dataSource.data = this.constraints;
+      });
+
+    this._userService.currentUserSettings.pipe(takeUntil(this._unsubscribe)).subscribe(settings => {
+      this.settings = settings;
+    });
+  }
+
+  ngOnDestroy(): void {
+    this._unsubscribe.next();
+    this._unsubscribe.complete();
+  }
+
+  isPaginatorVisible(): boolean {
+    return (
+      this.constraints &&
+      this.constraints.length > 0 &&
+      this.paginator &&
+      this.constraints.length > this.paginator.pageSize
+    );
+  }
+
+  isLoadingData(data: Constraint[]): boolean {
+    return _.isEmpty(data) && !this.isClusterRunning;
+  }
+
+  hasNoData(data: Constraint[]): boolean {
+    return _.isEmpty(data) && this.isClusterRunning;
+  }
+
+  toggleDetails(element: Constraint): void {
+    this.isShowDetails[element.name] = !this.isShowDetails[element.name];
+  }
+
+  listMatch(element: Kind[]): string {
+    const matches = [];
+    element.forEach(x => matches.push(x.kinds));
+    return matches.join(', ');
+  }
+
+  getViolationCount(violations: Violation[]): number {
+    return violations ? violations.length : 0;
+  }
+
+  add(): void {
+    const dialogConfig: MatDialogConfig = {
+      data: {
+        title: 'Add Constraint',
+        projectId: this.projectID,
+        clusterId: this.cluster.id,
+        mode: Mode.Add,
+        confirmLabel: 'Add',
+      },
+    };
+
+    this._matDialog
```

Comment:
This can be actually simplified to just `this._matDialog.open(ConstraintDialog, dialogConfig) since we do not do anything based on the dialog result. I am not sure why it is so common in our code if we are not using the subscribe at all.